### PR TITLE
8343317: Extend test generation tool to handle APX NDD/NF flavor of instructions

### PR DESCRIPTION
**Pull request by hanklo6**
Extend test generation tool and gtest (added as part of https://bugs.openjdk.org/browse/JDK-8339507) to handle Intel APX NDD/NF instructions.

Intel Advanced Performance Extensions (APX) adds non destructive destination (NDD) and no flags (NF) flavor for the scalar instructions through EVEX encoding. For more information about APX, see https://www.intel.com/content/www/us/en/developer/articles/technical/advanced-performance-extensions-apx.html.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8343317](https://bugs.openjdk.org/browse/JDK-8343317): Extend test generation tool to handle APX NDD/NF flavor of instructions (**Sub-task** - P4)

### Reviewers
 * [Sandhya Viswanathan](https://openjdk.org/census#sviswanathan) (@sviswa7 - **Reviewer**)
 * [Srinivas Vamsi Parasa](https://openjdk.org/census#sparasa) (@vamsi-parasa - Author)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/21795/head:pull/21795` \
`$ git checkout pull/21795`

Update a local copy of the PR: \
`$ git checkout pull/21795` \
`$ git pull https://git.openjdk.org/jdk.git pull/21795/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 21795`

View PR using the GUI difftool: \
`$ git pr show -t 21795`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/21795.diff">https://git.openjdk.org/jdk/pull/21795.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/21795#issuecomment-2448511857)
</details>

### CODE REVIEW

File: test/hotspot/gtest/x86/test_assemblerx86.cpp
Comment (vamsi-parasa) [inline test/hotspot/gtest/x86/test_assemblerx86.cpp:1]: [Optional] In lines 53 and 58, it prints "Ours" and "Theirs". This is vague. However, it looks like this is the convention used in test_assembler_aarch64.cpp as well.

Comment (hanklo6) [reply test/hotspot/gtest/x86/test_assemblerx86.cpp:1]: Thanks, I updated the label.

Comment (vamsi-parasa) [reply test/hotspot/gtest/x86/test_assemblerx86.cpp:1]: Thanks Hank!

**Review by sviswa7 — APPROVED** [review]
Looks good to me.

**Review by vamsi-parasa — APPROVED** [review]
This PR looks good to me. 

Also, I Independently tested that this tool works (i.e. all the tests are passing) after pulling in the APX encoding bugfix (https://github.com/openjdk/jdk/pull/21770) changes recently integrated into OpenJDK.

Thanks,
Srinivas Vamsi Parasa (from Intel)

**Comment by sviswa7** [comment]
@vnkozlov Could you please run this PR through your testing?

**Pull request closed without merging**